hooks/queries/votes/usePolymarketData.ts

```diff
@@ -0,0 +1,57 @@
+import { useQuery } from "@tanstack/react-query";
+
+type PolymarketResponse = {
+  slug: string;
+  title: string;
+  icon: string;
+  description: string;
+  startDate: string;
+  creationDate: string;
+  endDate: string;
+  volume: number;
+  createdAt: string;
+  updatedAt: string;
+}[];
+
+export type PolymarketData = PolymarketResponse[number] & {
+  link: string;
+};
+
+/**
+ * Searches Polymarket's api for a a market using a market's title.
+ *
+ * @param {string} title - The title of the market.
+ * @returns {PolymarketData} data - Relevant market data
+ */
+async function getPolymarketData(title: string): Promise<PolymarketData> {
+  const POLYMARKET_BASE_URL = "https://polymarket.com";
+  const params = new URLSearchParams({
+    _c: "all",
+    _p: "1",
+    _q: title,
+  });
+
+  const url = `https://polymarket.com/api/events/search?${params.toString()}`;
+  const res = await fetch(url);
+  const search = (await res.json()) as PolymarketResponse;
+  if (!search.length) {
+    throw new Error("Unable to find market.");
+  }
+  const data = search[0];
+
+  return {
+    ...data,
+    link: `${POLYMARKET_BASE_URL}/event/${data.slug}`,
```

Comment:
you might just want to verify that data.slug exists, or just throw an error if it doesnt, since you are casting the response